### PR TITLE
move graph APIs into separate class in order to be reused

### PR DESCRIPTION
**Pull request by sophiaso**
**Updated**: Some methods are moved from  `AADAuthenticationFilter.java` to `AzureADGraphClient.java`, in order to be reused later in spring-security-5 oauth2 update.

### Starter Names
  - active directory spring boot starter

### CODE REVIEW

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java [hunk 1]
```diff
@@ -99,21 +60,21 @@ protected void doFilterInternal(
...
                 final ServiceEndpoints serviceEndpoints =
                         serviceEndpointsProp.getServiceEndpoints(aadAuthFilterProp.getEnvironment());
+                final AzureADGraphClient client = new AzureADGraphClient(aadAuthFilterProp, serviceEndpointsProp);
 
                 if (principal == null || graphApiToken == null || graphApiToken.isEmpty()) {
                     principal = new UserPrincipal(idToken, serviceEndpoints);
-                    graphApiToken = acquireTokenForGraphApi(
-                            idToken, principal.getClaim().toString(), serviceEndpoints).getAccessToken();
+                    graphApiToken =
+                            client.acquireTokenForGraphApi(idToken, principal.getClaim().toString()).getAccessToken();
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java]: about line wrap, how about a local variable to hold `principal.getClaim().toString()` and pass to acquire for avoiding line wrap ?

Comment (sophiaso) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilter.java]: Yes, I  think it's better and also better readability, will change.

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java [hunk 2]
```diff
@@ -46,4 +80,73 @@ private static String getResponseStringFromConn(HttpURLConnection conn) throws I
...
+    }
+
+    private List<UserGroup> loadUserGroups(String graphApiToken) throws IOException {
+        final String responseInJson = getUserMembershipsV1(graphApiToken);
+        final List<UserGroup> lUserGroups = new ArrayList<>();
+        final ObjectMapper objectMapper = JacksonObjectMapperFactory.getInstance();
+        final JsonNode rootNode = objectMapper.readValue(responseInJson, JsonNode.class);
+        final JsonNode valuesNode = rootNode.get("value");
+        int i = 0;
+        while (valuesNode != null
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java]: seems i only appears in loop, how about `for (int i; condition; i++)`

Comment (sophiaso) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java]: Thanks for the remind, will update.

**Comment by sophiaso** [comment]
Updated according to the comments.

**Pull request merged**